crates/solver/src/solver/http_solver.rs

```diff
@@ -367,6 +370,20 @@ impl Solver for HttpSolver {
         if orders.is_empty() {
             return Ok(Vec::new());
         };
+        let prices = external_prices.clone().into_http_solver_prices();
+        orders.extend(
+            liquidity
+                .iter()
+                .filter_map(|liquidity| match liquidity {
+                    Liquidity::LimitOrder(order) => Some(order.clone()),
+                    _ => None,
+                })
+                .filter(|order| {
+                    prices.get(&order.sell_token).is_some()
+                        && prices.get(&order.buy_token).is_some()
```

Comment:
I'm not entirely sure this is needed because of:
```rust
        // We are guaranteed to have price estimates for all tokens that are relevant to the
        // objective value by the driver. It is possible that we have AMM pools that contain tokens
        // that are not any order's tokens. We used to fetch these extra prices but it would often
        // slow down the solver and the solver can estimate them on its own.
        let price_estimates = external_prices.into_http_solver_prices();
```
But when I passed all orders I always got [settlement simulation failures](https://dashboard.tenderly.co/gp-v2/staging/simulator/72e65804-2bf8-4b5b-ad52-dc6b637a5b6f). I think the error is always caused by some transfer exceeding the balance of our settlement contract. I'm not sure how filtering out the orders without prices affects that but it seems to be related.
Any ideas on that would be greatly appreciated.